### PR TITLE
raml-tester/issues/76

### PR DESCRIPTION
**Pull request by arielsegura**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -75,15 +75,23 @@
...
-                        <configuration>
-                            <classified>true</classified>
-                        </configuration>
+                        <!--<goals>-->
+                            <!--<goal>backport7to6</goal>-->
+                        <!--</goals>-->
+                        <!--<configuration>-->
+                            <!--<classified>true</classified>-->
+                        <!--</configuration>-->
                     </execution>
```
Comment (ddossot) [inline pom.xml]: Why commenting XML? If it's not needed, remove it. If it's maybe needed, put it in a profile.

**Comment by ddossot** [comment]
@arielsegura build fails for this PR...

**Comment by arielsegura** [comment]
Yes, sorry, this is not intended to be merged right now. I just wanted to show another approach for https://github.com/nidi3/raml-tester/issues/76

File: pom.xml (hunk 1 above)
Comment (arielsegura) [inline pom.xml]: Because my branch depends on java 8 and that plugin won't work. It makes java 7 code work with java 6.

Comment (ddossot) [inline pom.xml]: I realize that, what I didn't know was that this PR is not intended for merging (as you said elsewhere). Thus the "draft" aspect of it.

**Comment by nidi3** [comment]
Thanks for this pull request.

There were 2 reasons for my implementation:
- Keep `RamlViolations` an `Iterable<String>` because of backwards compatibility. But it's still version 0.x so why not change things.
- Use `Object` because a (java) consumer has to do a cast anyways, even if we use an interface. You added a `asJson` method which mitigates to problem. But JSON is just one (of many) serialization formats. But serialization is not a direct concern of ViolationMessage.

So I'd use your implementation without `asJson` but with ObjectMapper friendly POJOs.

**Comment by arielsegura** [comment]
We can still keep `RamlViolations` as `Iterable<String>`, if you think it's more convenient. It shouldn't be a big deal:
```
   public List<String> asList() {
        return ramlViolationMessages.stream()
                .map(ramlViolationMessage -> ramlViolationMessage.getMessage())
                .collect(Collectors.toList());
    }
    @Override
    public Iterator<String> iterator() {
        return asList().iterator();
    }
```

I agree on the serialization thing. Instead of having `asJson()`
We could do something like:
```
public interface RamlViolationCause {
    RamlViolationCauseContent getMessage();
}
```
Where `RamlViolationCauseContent` (don't like this name though) is an abstract class used/extended accordingly, depending on the situation. I mean, each violation has a different "content". For instance, XML errors usually make reference to a `line` and a `column` whereas JSON(schema) errors refers to particular properties. 

```
public abstract class RamlViolationCauseContent {
    String message;
    String detail;

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getDetail() {
        return detail;
    }

    public void setDetail(String detail) {
        this.detail = detail;
    }
}
```
Does this make sense for you?

**Pull request closed without merging**